### ISSUE TITLE
maven cli wrapper plugin

### ISSUE THREAD

**Issue by ennerf** [sass/sass-site#660]
Reading through the documentation I wasn't sure how to best integrate with Maven, so I wrote a small plugin that downloads and calls the CLI: 

[sass-cli-maven-plugin](https://github.com/HebiRobotics/sass-cli-maven-plugin)

Feel free to add it to the docs, or ignore if there is already a better way to do it.

**Comment by Goodwine** [comment]
There was a Maven plugin for libsass but not for dart-sass, thanks for letting us know, I'll add a link shortly :)

**Comment by melloware** [comment]
There is a full proper wrapper now using Dart Sass with Maven or Gradle.

https://github.com/cleydyr/dart-sass-maven-plugin

**Comment by melloware** [comment]
I am pretty sure the one I posted supports all 3 OS Windows Linux MacOS and knows also has a Sass Watch properly integrated.  I didn't study the first one in detail to compare the two but the latter emulates was the old lib sass plug-in did almost exactly.

**Comment by ennerf** [comment]
> the one I posted supports all 3 OS Windows Linux MacOS and knows which you are using. also has a Sass Watch properly integrated.

The first one does as well. I had a quick look at the source code and I think the main differences are:

**sass-cli-maven-plugin**
* supports arbitrary versions of the dart plugin and downloads the specified version for the host OS
* cli arguments are specified as an`<args>` list and passed to the executable, e.g., `<arg>--embedd-sources</arg>`
* currently the only argument with special support is `watch`, but it'd be fairly straight forward to add maven parameter wrappers
* the reasoning behind this approach was to support new versions/features immediately without requiring a plugin release

**dart-sass-maven-plugin**
* bundles a fixed version of dart-sass in the jar and extracts it at runtime
* the arguments are exposed as maven parameters, i.e., `<embedSources>true</<embedSources>`
* the bundled approach requires matching plugin releases for new sass-dart releases

Basically one is a thin wrapper around the CLI executable, and the other one is an abstraction on top that hides the actual calls.  It looks like the dart-sass plugin was started a bit earlier, so I wouldn't have written another one if I had been aware of it. That being said, I don't like that the latest plugin release is already at least 9 months worth of dart-sass releases out of date.

**Comment by melloware** [comment]
That is a good summary that one is a thin CLI wrapper the other is an abstraction so users don't have to know anything about Dart SASS.  I do agree the included DART executables is nice for not needing an internet connection but does make waiting for releases a pain.  I like the way the FrontEnd NPM plugin works where you say what version you want it it downloads and caches it.   

Maybe you two plugin developers can work together to make 1 great Maven plugin for Dart SASS.

**Comment by ennerf** [comment]
> I like the way the FrontEnd NPM plugin works where you say what version you want it it downloads and caches it.

This is how the `sass-cli` plugin works. Versioned executables get cached and are only [downloaded on first execution](https://github.com/HebiRobotics/sass-cli-maven-plugin/blob/main/src/main/java/us/hebi/sass/RunMojo.java#L104-L114).

> Maybe you two plugin developers can work together to make 1 great Maven plugin for Dart SASS.

The cli-plugin already exposes all features of dart-sass, so it's mostly about providing some maven wrappers for auto-complete, and maybe some added documentation in the Readme. Can you create an issue on the repo about what you're actually missing? Many parameters, e.g., `<noCharset>` are probably used so infrequently that it hardly seems worth it to add wrappers that could potentially break in the future.

**Comment by nex3** [comment]
@Goodwine Can you add the other plugin to the list with a summary of the differences?
